lang/java/reef-tang/tang/src/main/java/org/apache/reef/tang/implementation/types/ConstructorDefImpl.java

```diff
@@ -129,6 +130,48 @@ public boolean equals(final Object o) {
     return equalsIgnoreOrder((ConstructorDef<?>) o);
   }
 
+  public boolean equalsNew(final Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    int length = getArgs().length;
+    ConstructorDef<?> def = (ConstructorDef<?>) o;
+    if (length != def.getArgs().length) {
+      return false;
+    } else if (length == 0) {
```

Comment:
I think we don't need `else`'s after `if` which ends in `return`. Separate `if`'s would be more readable